lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -12,39 +12,47 @@
   mk_resource_methods
   
   def create
-    inst_cmd = "New-WebAppPool -Name \"#{@resource[:name]}\" -ErrorAction Stop"
-    result   = self.class.run(inst_cmd)
+    cmd = []
+    cmd << self.add_powershell2_support if self.ps_major_version == 2
+    cmd << "New-WebAppPool -Name \"#{@resource[:name]}\" -ErrorAction Stop\n"
+
+    inst_cmd = cmd.join
+    result = self.class.run(inst_cmd)
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:exitcode] == 0
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:errormessage].nil?
   end
 
   def update
-    inst_cmd = []
-
-    inst_cmd << self.class.ps_script_content('_setapppool', @resource)
+    cmd = []
+    cmd << self.add_powershell2_support if self.ps_major_version == 2
```

Comment:
Instead of remembering to add this line to each method, would it be better to add it to the `run` method? Inside run is where execution logic is, and is only one place to edit if we need to change this where we will have dozens of providers.

Something like:

```
cmd = []
if self.ps_major_version == 2
  cmd << self.add_powershell2_support
  cmd << command
else
  cmd = command
end
```